node/client/src/lib.rs

```diff
@@ -391,4 +384,39 @@ impl Client {
         let index = self.beacon_chain.chain.best_block().index() + 1;
         self.get_uid_to_authority_map(index).1
     }
+
+    /// Fetch "coupled" blocks by hash.
+    pub fn fetch_blocks(&self, hashes: Vec<CryptoHash>) -> Result<Vec<(SignedBeaconBlock, SignedShardBlock)>, String> {
+        let mut result = vec![];
+        for hash in hashes.iter() {
+            match self.beacon_chain.chain.get_block(&BlockId::Hash(*hash)) {
+                Some(beacon_block) => {
+                    let shard_block = self.shard_client.chain.get_block(&BlockId::Hash(beacon_block.body.header.shard_block_hash)).expect(BEACON_SHARD_BLOCK_MATCH);
+                    result.push((beacon_block, shard_block));
+                },
+                None => return Err(format!("Missing {:?} in beacon chain", hash))
+            }
+        }
+        Ok(result)
+    }
+
+    /// Fetch "coupled" blocks by index range.
+    pub fn fetch_blocks_range(&self, from_index: u64, til_index: u64) -> Result<Vec<(SignedBeaconBlock, SignedShardBlock)>, String> {
+        let mut result = vec![];
+        for i in from_index..=til_index {
+            match self.beacon_chain.chain.get_block(&BlockId::Number(i)) {
```

Comment:
`get_block` accepts `BlockId::Hash`. The hash is the same for shard and beacon block, I think.
After we decouple beacon and shard the whole API of the client will likely change anyway.

Reply:
hm, unclear how "`get_block` accepts `BlockId::Hash`." relevant here.

when we decouple - yeah will change.